searx/plugins/hostname_replace.py

```diff
@@ -17,16 +17,30 @@
 
 logger = logger.getChild(plugin_id)
 parsed = 'parsed_url'
+_url_fields = ['iframe_src', 'audio_src']
 
 
 def on_result(request, search, result):
-    if parsed not in result:
-        return True
+
     for (pattern, replacement) in replacements.items():
-        if pattern.search(result[parsed].netloc):
-            if not replacement:
-                return False
-            result[parsed] = result[parsed]._replace(netloc=pattern.sub(replacement, result[parsed].netloc))
-            result['url'] = urlunparse(result[parsed])
+
+        if parsed in result:
+            if pattern.search(result[parsed].netloc):
+                # to keep or remove this result from the result list depends
+                # (only) on the 'parsed_url'
+                if not replacement:
+                    return False
+                result[parsed] = result[parsed]._replace(netloc=pattern.sub(replacement, result[parsed].netloc))
+                result['url'] = urlunparse(result[parsed])
+
+        if not replacement:
```

Comment:
> If in the future, this is attribute is set , this condition is going to prevent the hostname replacement in the iframe_src attribute.

Not sure I fully understand you .. but:

- if there is no replacement (``if not replacement``)  we can't replace a URL in the iframe
- the hostname_replace plugin can only drop results from the result list, but it can't drop an iframe content.

Reply:
I rewrote this commit, see 795e8af6